### PR TITLE
added override values for helm feature

### LINKED ISSUE #901: Override helm chart default values
**Is your feature request related to a problem? Please describe.**</br>
 > An improvement over the existing workaround

**Describe the solution you'd like.**</br>
 > Kubescape should provide a way to override default values when scanning helm charts. 
> Since a given directory can contain multiple charts, the argument should accept a mapping of `chart -> values file`.
> For example: 
> `kubescape scan <directory> --helm-value-file path/to/chart1:path/to/values.yaml`
>
> Overriding multiple charts is possible by a comma-separated value: 
> `--helm-value-file path/to/chart1:path/to/values1.yaml,path/to/chart2:path/to/values2.yaml`

**Describe alternatives you've considered.**</br>
 > A workaround exists but it involves two steps- 
> 1. Render the helm chart using the `helm template` command (with `--set` to override values)
> 2. Run `kubescape scan` on the rendered yaml

**Additional context.**</br>
 > Add any other context or screenshots about the feature request here.

### PR DESCRIPTION
**Pull request by AvineshTripathi**
fixes #901

With merge of this PR kubescape will support a flag `--helm-value-file` which takes input of path/paths of value file seperated by `,` and override the default value file of helm.

EDITED : 

now the flag takes a string value seperated by space e.g. 
`--helm-value-file=path/to/chart1:path/to/value/file1,path/to/value/file2 path/to/chart2:path/to/value/path1....."  `

Approach:

The working of this feature is similiar to what `helm/helm` repo handles overriding however due to different structures I could not use those functions. ref https://github.com/helm/helm/blob/main/pkg/cli/values/options.go#L42

### CODE REVIEW

**Comment by AvineshTripathi** [comment]
Update: fixing test cases

**Comment by dwertent** [comment]
@AvineshTripathi the checks failed

File: core/cautils/helmchart.go [hunk 1]
```diff
@@ -90,3 +112,23 @@ func (hc *HelmChart) GetWorkloads(values map[string]interface{}) (map[string][]w
 	}
 	return workloads, errs
 }
+
+// code from helm/helm repo under pkg/cli/values/option.go
+func mergeMaps(a, b map[string]interface{}) map[string]interface{} {
```
Comment (dwertent) [inline core/cautils/helmchart.go:117]: Please add some unitests for this function

**Review by dwertent — CHANGES_REQUESTED** [review]
I'm missing the mapping between the chart and the values file.
e.g. A directory with multiple helm charts and you want to override one of the values file, how will you do it?

**Comment by AvineshTripathi** [comment]
> I'm missing the mapping between the chart and the values file. e.g. A directory with multiple helm charts and you want to override one of the values file, how will you do it?

can you explain the example in more detail please?

**Comment by AvineshTripathi** [comment]
@dwertent ^^

**Comment by dwertent** [comment]
@AvineshTripathi I'm waiting for the check to pass before I review :)

**Comment by AvineshTripathi** [comment]
Hey @dwertent thanks for the response however I am facing a lot trouble with conflicts(shared in slack as well).
To explain more,

I did changes and pushed to PR but it showed conflicts in a file which was not changed by my PR later as I resolved the conflicts build started throwing error in the same file of resolved conflicts. This situation has been constant for me even after resolving conflicts in previous PR as well so I need some help with it

**Comment by AvineshTripathi** [comment]
Hey folks I tried again resolving the conflicts still things remain same so I would need some help in resolving the conflicts. The branch is working perfectly when run locally

**Comment by matthyx** [comment]
@AvineshTripathi do you still need help with conflicts?

**Comment by matthyx** [comment]
@AvineshTripathi do we want to revisit this?

**Comment by dwertent** [comment]
Stale PR

**Pull request closed without merging**